### PR TITLE
Renewal Date earlier than Original Issue Date #352

### PR DESCRIPTION
**Pull request by dan9000**
Adjust Drools rules to compare renewal date to original issue date instead
of the default earliest acceptable date 01-01-1900

### CODE REVIEW

**Comment by dan9000** [comment]
Do we want to disallow a renewal date that's before today's date too? This is also mentioned in #352

**Comment by jasonaowen** [comment]
Jenkins, test this please.

**Comment by cecilia-donnelly** [comment]
Jenkins, test this please.

**Comment by cecilia-donnelly** [comment]
@dan9000, would you mind squashing this into one commit for adding tests and one for updating the rules?  We try to avoid merge commits in PRs, and I find having two commits with the title string "License renewal date validation" a bit confusing, even though they have different full messages.

**Comment by dan9000** [comment]
@cecilia-donnelly I squashed the commits

File: psm-app/integration-tests/src/test/java/gov/medicaid/features/enrollment/ui/LicenseInfoPage.java [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package gov.medicaid.features.enrollment.ui;
 
 import gov.medicaid.features.PsmPage;
+import org.openqa.selenium.WebElement;
```
Comment (jasonaowen) [inline psm-app/integration-tests/src/test/java/gov/medicaid/features/enrollment/ui/LicenseInfoPage.java]: It looks like this [import is unused](https://ci.psm.solutionguidance.com/job/checkstyle/153/checkstyleResult/new/), according to `./gradlew checkstyleMain checkstyleTest`. Would you please remove it?

File: psm-app/integration-tests/src/test/java/gov/medicaid/features/enrollment/ui/LicenseInfoPage.java [hunk 2]
```diff
@@ -11,16 +12,19 @@
...
+    private static final String RENEWAL_DATE_ERROR_MESSAGE =
+            "cannot be earlier than Original Issue Date";
+
     public void clickNo() {
         click($("input[value='N'"));
     }
 
     public void addLicense() {
-        click($("#addLicense"));
+        click($("#addLicense, #addFacilityLicense"));
```
Comment (jasonaowen) [inline psm-app/integration-tests/src/test/java/gov/medicaid/features/enrollment/ui/LicenseInfoPage.java]: Super neat! I didn't know you could give it alternatives like this.

**Comment by jasonaowen** [comment]
Jenkins, test this please.

File: psm-app/cms-business-process/src/main/resources/cms.validation.drl [hunk 3]
```diff
@@ -7603,21 +7604,22 @@ dialect 'mvel'
...
-rule 'License Renewal Date Cannot Be Before 01-01-1900'
+rule 'License Renewal Date Cannot Be Before Original Issue Date'
 dialect 'mvel'
     when
         LookupEntry(type == "FieldGroup", value == UISection.LICENSE_INFORMATION.value())
         $provider: ProviderInformationType()
         LicenseInformationType( $licenses: license ) from $provider.licenseInformation
-        $license: LicenseType($renewalDate : renewalDate, renewalDate != null) from $licenses
-        eval($renewalDate.before(getEarliestAcceptableDate()))
+        $license: LicenseType($renewalDate : renewalDate, renewalDate != null,
```
Comment (cecilia-donnelly) [inline psm-app/cms-business-process/src/main/resources/cms.validation.drl]: Looks like there's an extra space here, but apparently doesn't harm anything.

File: psm-app/integration-tests/src/test/java/gov/medicaid/features/enrollment/steps/EnrollmentStepDefinitions.java [hunk 4]
```diff
@@ -16,13 +19,17 @@
 
     private EnrollmentPage enrollmentPage;
     private OwnershipInfoPage ownershipInfoPage;
+    private LicenseInfoPage licenseInfoPage;
 
     @Given("^I am entering ownership information$")
-    public void i_am_entering_ownership_information() {
+    public void i_am_entering_ownership_information() throws IOException {
```
Comment (cecilia-donnelly) [inline psm-app/integration-tests/src/test/java/gov/medicaid/features/enrollment/steps/EnrollmentStepDefinitions.java]: Where does the new IOException come from?

File: psm-app/integration-tests/src/test/java/gov/medicaid/features/enrollment/steps/EnrollmentSteps.java [hunk 5]
```diff
@@ -145,21 +155,36 @@ public void checkForTooYoungError() throws Exception {
...
     public void enterNotAProviderAtPublicHealthServiceIndianHospital() {
         licenseInfoPage.clickNo();
     }
 
     @Step
-    public void enterIndividualLicenseInfo() {
+    public void enterLicenseInfo() {
         licenseInfoPage.addLicense();
-        licenseInfoPage.addLicenseType(LICENSE_TYPE);
+        licenseInfoPage.addLicenseType(licenseType);
```
Comment (cecilia-donnelly) [inline psm-app/integration-tests/src/test/java/gov/medicaid/features/enrollment/steps/EnrollmentSteps.java]: Nice.

File: psm-app/integration-tests/src/test/java/gov/medicaid/features/enrollment/steps/EnrollmentStepDefinitions.java (hunk 4 above)
Comment (dan9000) [reply psm-app/integration-tests/src/test/java/gov/medicaid/features/enrollment/steps/EnrollmentStepDefinitions.java]: On upload license. The old tests used a Dental Clinic provider type that didn't require licensing, so I switched it to Head Start

**Comment by jasonaowen** [comment]
> One thing: this rule doesn't appear to be triggered when I click "Save as Draft."

@cecilia-donnelly It looks like this was an intentional choice on the part of the previous developers: ["validate when saving the first page"](https://github.com/SolutionGuidance/psm/blob/e147ebe5f6f2c70943962ba20a1e92f5f6e11f47/psm-app/cms-web/src/main/java/gov/medicaid/controllers/EnrollmentPageFlowController.java#L1028-L1033). If we want to change that, we should do it outside of this PR, I think.

**Comment by cecilia-donnelly** [comment]
Thanks for tracking that down, @jasonaowen!

**Pull request merged**